share/functions/__fish_config_interactive.fish

```diff
@@ -22,42 +22,54 @@ function __fish_config_interactive -d "Initializations that should be performed
         set userdatadir $XDG_DATA_HOME
     end
 
+    if not set -q fish_greeting
+        set -l line1 (printf (_ 'Welcome to fish, the friendly interactive shell' ))
+        if not set -q __fish_init_2_3_0
+            set -l line2 \n(printf (_ 'Type %shelp%s for instructions on how to use fish %s') (set_color green) (set_color normal))
+        else
+            set -l line2 ''
+        end
+        set -U fish_greeting $line1$line2
+    end
+
     #
     # If we are starting up for the first time, set various defaults
-    #
-    if not set -q __fish_init_1_50_0
-        if not set -q fish_greeting
-            set -l line1 (printf (_ 'Welcome to fish, the friendly interactive shell') )
-            set -l line2 (printf (_ 'Type %shelp%s for instructions on how to use fish') (set_color green) (set_color normal))
-            set -U fish_greeting $line1\n$line2
+    #    
+    if not set -q __fish_init_2_39_8 # bump this to 2_4_0 when rolling release if anything changes after 9/10/2016
+        set -g colors_backup "$HOME/fish_previous_colors-(date).txt"
```

Comment:
Not only that but even if we supported command substitution inside quoted strings it is generally a bad idea to use the user locale for formatting the date in this type of context. It's almost always a better to use a fixed format such as ISO 8601. I'll take care of it.